lib/datatrans/config.rb

```diff
@@ -75,6 +75,11 @@ def web_transaction(*args)
     end
 
     def xml_transaction(*args)
+      message = <<-WARNING
+      [WARNING] Usage of XML API is deprecated by Datatrans. Consider moving to JSON API until June 3rd, 2024.
+      WARNING
+
+      Warning.warn(message)
```

Comment:
Not sure about adding this. Might cause some issues for some people depending on how they set up their monitoring.
I wouldn't add this just to make sure no existing code cannot possibly break.
Chances that this breaks anyone's code is slim, but who knows...
https://xkcd.com/1172/

Also, I don't think it's the job of this gem to tell people to use the new API, it just enables using it. I think adding that message in the changelog is good enough.